### DISCUSSION TITLE
Run a script on button press

### DISCUSSION THREAD

**Discussion by Luke-From-IT** [IBM/mac-ibm-notifications#26] [General]
Hey, we're looking at using this to provide notifications to our staff about critical updates to certain software.

We can use x-apple.systempreferences to open System Preference panes, but would like to also run a shell/python script on a secondary button press.

Is this doable with the -link implementation? Or are links only URL based?

I think once a script is runable, then the possibilities are endless.

**Comment by SMartorelli** [comment]
Hi @luke-carwowo, the `link` CTA Type is URL based. We have not implemented a `script` CTA Type on purpose since we don't want the App to be able to run any script for security reasons. If you're running the notification through a script you can simple parse the exit code from the App and run another portion of the script based on it. The app returns a different exit code for every possible destructive user interaction on the UI, for example the "Main button click" action returns `0`, the "Secondary button click" returns `2` and so on. You'll find all the possible exit codes from the App in the [Usage wiki page](https://github.com/IBM/mac-ibm-notifications/wiki/Usage#return-values).
